### PR TITLE
Issue 18132 content second lang not included site search

### PR DESCRIPTION
**Pull request by dsilvam**
- When rendering a page's html, the content version info was being first looked up in the given language, and after not being found, the idea was to look for it in the default language, but the condition was not ok. It was just checking the contentInfo being null and not also its properties being null.

- A new integration test was added. 
- Utility methods for publish containers and templates were added to the respective dataGen classes.

### CODE REVIEW

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.exception.WebAssetException;
+import com.dotmarketing.factories.PublishFactory;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.folders.model.Folder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotcms.datagen.ContentTypeDataGen;
+import com.dotcms.datagen.ContentletDataGen;
+import com.dotcms.datagen.FolderDataGen;
+import com.dotcms.datagen.HTMLPageDataGen;
+import com.dotcms.datagen.LanguageDataGen;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.datagen.TemplateDataGen;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.ContainerStructure;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.beans.ContainerStructure'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        ContentletDataGen.publish(urlMappedContent);
+
+        final MultiTree multiTree = new MultiTree(htmlPage.getIdentifier(),
+                container.getIdentifier(),
+                urlMappedContent.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+
+        APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+
+        String html = APILocator.getHTMLPageAssetAPI().getHTML(htmlPage.getURI(), site, true,
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Local variable 'html' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotmarketing.factories.PublishFactory;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.UUIDGenerator;
+import java.util.Collections;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Collections'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+import static com.dotcms.rendering.velocity.directive.ParseContainer.getDotParserContainerUUID;
+import static com.dotmarketing.util.Constants.USER_AGENT_DOTCMS_SITESEARCH;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.ContentTypeBuilder;
+import com.dotcms.datagen.ContainerDataGen;
+import com.dotcms.datagen.ContentTypeDataGen;
+import com.dotcms.datagen.ContentletDataGen;
+import com.dotcms.datagen.FolderDataGen;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.FolderDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
+package com.dotmarketing.portlets.htmlpages.business;
+
+import static com.dotcms.rendering.velocity.directive.ParseContainer.getDotParserContainerUUID;
+import static com.dotmarketing.util.Constants.USER_AGENT_DOTCMS_SITESEARCH;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.ContentTypeBuilder;
+import com.dotcms.datagen.ContainerDataGen;
+import com.dotcms.datagen.ContentTypeDataGen;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.ContentTypeDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.UUIDGenerator;
+import java.util.Collections;
+import java.util.Date;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Date'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class HTMLPageAssetAPIImplTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.ContainerStructure;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.exception.WebAssetException;
+import com.dotmarketing.factories.PublishFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.factories.PublishFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
+package com.dotmarketing.portlets.htmlpages.business;
+
+import static com.dotcms.rendering.velocity.directive.ParseContainer.getDotParserContainerUUID;
+import static com.dotmarketing.util.Constants.USER_AGENT_DOTCMS_SITESEARCH;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.ContentTypeBuilder;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.contenttype.model.type.ContentTypeBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.exception.WebAssetException;
+import com.dotmarketing.factories.PublishFactory;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Constants;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.util.Constants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.util.Constants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.folders.model.Folder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.contenttype.model.type.ContentTypeBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.beans.ContainerStructure'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Local variable 'html' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.FolderDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.factories.PublishFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Date'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.ContentTypeDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Collections'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.UUIDGenerator;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class HTMLPageAssetAPIImplTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)

**Pull request merged**